bundles/net.tourbook/src/net/tourbook/ui/views/tourDataEditor/TourDataEditorView.java

```diff
@@ -9005,21 +8951,14 @@ private void updateUI_Time(final Widget widget) {
       /*
        * check if a time control is currently used
        */
-      if ((widget == _deviceTime_Elapsed._spinHours
-            || widget == _deviceTime_Elapsed._spinMinutes
-            || widget == _deviceTime_Elapsed._spinSeconds
-            || widget == _deviceTime_Recorded._spinHours
-            || widget == _deviceTime_Recorded._spinMinutes
-            || widget == _deviceTime_Recorded._spinSeconds
-            || widget == _deviceTime_Paused._spinHours
-            || widget == _deviceTime_Paused._spinMinutes
-            || widget == _deviceTime_Paused._spinSeconds
-            || widget == _computedTime_Moving._spinHours
-            || widget == _computedTime_Moving._spinMinutes
-            || widget == _computedTime_Moving._spinSeconds
-            || widget == _computedTime_Break._spinHours
-            || widget == _computedTime_Break._spinMinutes
-            || widget == _computedTime_Break._spinSeconds) == false) {
+      if (((widget != _deviceTime_Elapsed._spinHours) && (widget != _deviceTime_Elapsed._spinMinutes) && (widget != _deviceTime_Elapsed._spinSeconds)
+            && (widget != _deviceTime_Recorded._spinHours)
+            && (widget != _deviceTime_Recorded._spinMinutes) && (widget != _deviceTime_Recorded._spinSeconds)
+            && (widget != _deviceTime_Paused._spinHours) && (widget != _deviceTime_Paused._spinMinutes)
+            && (widget != _deviceTime_Paused._spinSeconds) && (widget != _computedTime_Moving._spinHours)
+            && (widget != _computedTime_Moving._spinMinutes) && (widget != _computedTime_Moving._spinSeconds)
+            && (widget != _computedTime_Break._spinHours) && (widget != _computedTime_Break._spinMinutes)
+            && (widget != _computedTime_Break._spinSeconds))) {
```

Comment:
This new code is just horror for me

-> disable your automatic optimization or review it

![image](https://user-images.githubusercontent.com/1283445/123043840-2b32d080-d3f9-11eb-82e2-0e4828b28737.png)

Reply:
I think it's because of the new 4.20 code optimizations. Will fix

Reply:
Fixed